src/angular/10-updating-service-params/restaurant.component-httpparams.ts

```diff
@@ -54,48 +57,49 @@ export class RestaurantComponent implements OnInit, OnDestroy {
     this.onDestroy$.complete();
   }
 
-  createForm(): FormGroup {
-    return this.fb.group({
+  createForm(): FormGroup<{
+    state: FormControl<string>;
+    city: FormControl<string>;
+  }> {
+    return this.fb.nonNullable.group({
       state: { value: '', disabled: true },
       city: { value: '', disabled: true },
     });
   }
 
   onChanges(): void {
-    let state: string = this.form.get('state')?.value;
+    let state: string = this.form.controls.state.value;
 
-    this.form
-      .get('state')
-      ?.valueChanges.pipe(takeUntil(this.onDestroy$))
+    this.form.controls.state.valueChanges
+      .pipe(takeUntil(this.onDestroy$))
       .subscribe((val) => {
         this.restaurants.value = [];
         if (val) {
           // only enable city if state has value
-          this.form.get('city')?.enable({
+          this.form.controls.city.enable({
             onlySelf: true,
             emitEvent: false,
           });
 
           // if state has a value and has changed, clear previous city value
           if (state !== val) {
-            this.form.get('city')?.patchValue('');
+            this.form.controls.city.setValue('');
```

Comment:
Would this be a good use case for showing how `.reset()` works on a non-nullable FormControl, rather than manually setting the value to an empty string?

Reply:
that's a great idea